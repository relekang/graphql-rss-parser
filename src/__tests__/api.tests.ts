import './mockAxios'
import { testGraphqlApi } from './utils'

const websiteUrl = 'https://rolflekang.com'
const url = 'https://rolflekang.com/feed.xml'

testGraphqlApi`
  feed(url: "${url}") { title }
`

testGraphqlApi`
  feed(url: "${url}") {
    title
    link
    entries {
      title
      link
      pubDate
    }
  }
`

testGraphqlApi`
  feed(url: "${url}", parser: FEEDPARSER) { title }
`

testGraphqlApi`
  feed(url: "${url}", parser: RSS_PARSER) { title }
`

testGraphqlApi`
  feed(url: "${url}", parser: FEEDME) { title }
`

testGraphqlApi`
  feed(url: "${url}", parser: RSS_TO_JSON) { title }
`

testGraphqlApi`
  feed(url: "https://rolflekang.com/feed.json", parser: JSON_FEED_V1) { title }
`

testGraphqlApi`
  feed(url: "${url}") { title, badField }
`

testGraphqlApi`
  feed(url: "https://rolflekang.com/testing-simple-graphql-services") { title }
`

testGraphqlApi`
  feed(url: "https://non-existing-domain.com") { title }
`

testGraphqlApi`
  findFeed(url: "${websiteUrl}") {
    link
  }
`

testGraphqlApi`
  feed(url: "${url}", startTime: "2020-01-01", endTime: "2020-10-31") {
    entries { title }
  }
`
