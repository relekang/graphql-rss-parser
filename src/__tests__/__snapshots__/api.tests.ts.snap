// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`
  feed(url: "https://non-existing-domain.com") { title }
 1`] = `
Object {
  "data": Object {
    "feed": null,
  },
  "errors": Array [
    Object {
      "error": Object {
        "code": "dns-lookup-error",
        "message": "Could not find domain",
      },
      "path": Array [
        "feed",
      ],
    },
  ],
}
`;

exports[`
  feed(url: "https://rolflekang.com/feed.json", parser: JSON_FEED_V1) { title }
 1`] = `
Object {
  "data": Object {
    "feed": Object {
      "title": "Writing by Rolf Erik Lekang",
    },
  },
}
`;

exports[`
  feed(url: "https://rolflekang.com/feed.xml") {
    title
    link
    entries {
      title
      link
      pubDate
    }
  }
 1`] = `
Object {
  "data": Object {
    "feed": Object {
      "entries": Array [
        Object {
          "link": "https://rolflekang.com/using-certbot-with-ansible",
          "pubDate": "2020-11-08T00:00:00.000Z",
          "title": "Using certbot with Ansible",
        },
        Object {
          "link": "https://rolflekang.com/ansible-handlers-in-loops",
          "pubDate": "2020-09-12T00:00:00.000Z",
          "title": "Using Ansible handlers in loops",
        },
        Object {
          "link": "https://rolflekang.com/serving-plain-text-with-nextjs",
          "pubDate": "2020-05-26T00:00:00.000Z",
          "title": "Serving text/plain for curl with Next",
        },
        Object {
          "link": "https://rolflekang.com/wireless-uplinks-with-unifi",
          "pubDate": "2020-05-03T00:00:00.000Z",
          "title": "Wireless uplinks with Unifi",
        },
        Object {
          "link": "https://rolflekang.com/using-git-commits-instead-of-stash",
          "pubDate": "2019-11-14T00:00:00.000Z",
          "title": "Using git commits instead of git stash",
        },
        Object {
          "link": "https://rolflekang.com/twitter-cards-for-gatsby-posts",
          "pubDate": "2019-02-26T00:00:00.000Z",
          "title": "Twitter cards for Gatsby posts",
        },
        Object {
          "link": "https://rolflekang.com/related-articles-with-gatsby",
          "pubDate": "2019-02-20T00:00:00.000Z",
          "title": "Related articles with Gatsby",
        },
        Object {
          "link": "https://rolflekang.com/creating-a-cli-with-reason-native",
          "pubDate": "2019-02-12T00:00:00.000Z",
          "title": "Creating a CLI with Reason native",
        },
        Object {
          "link": "https://rolflekang.com/testing-simple-graphql-services",
          "pubDate": "2017-05-28T00:00:00.000Z",
          "title": "Testing simple GraphQL services",
        },
        Object {
          "link": "https://rolflekang.com/lets-run-some-races",
          "pubDate": "2017-01-02T00:00:00.000Z",
          "title": "Let's run some races this year",
        },
        Object {
          "link": "https://rolflekang.com/software-development-on-an-ipad",
          "pubDate": "2017-01-01T00:00:00.000Z",
          "title": "Software development on an iPad",
        },
        Object {
          "link": "https://rolflekang.com/filtering-lint-errors",
          "pubDate": "2016-06-10T00:00:00.000Z",
          "title": "Filtering lint errors",
        },
        Object {
          "link": "https://rolflekang.com/interrail-summer",
          "pubDate": "2015-07-26T00:00:00.000Z",
          "title": "Interrail summer",
        },
        Object {
          "link": "https://rolflekang.com/writing-latex-in-atom",
          "pubDate": "2015-05-23T00:00:00.000Z",
          "title": "Writing (latex) in Atom",
        },
        Object {
          "link": "https://rolflekang.com/rmoq",
          "pubDate": "2014-12-30T00:00:00.000Z",
          "title": "rmoq - a request mock cache",
        },
        Object {
          "link": "https://rolflekang.com/django-nopassword-one-point-o",
          "pubDate": "2014-10-03T00:00:00.000Z",
          "title": "django-nopassword reached 1.0",
        },
        Object {
          "link": "https://rolflekang.com/moving-to-pelican",
          "pubDate": "2014-05-07T00:00:00.000Z",
          "title": "Moving the blog to pelican",
        },
        Object {
          "link": "https://rolflekang.com/building-tumblr-theme-with-grunt",
          "pubDate": "2014-02-02T00:00:00.000Z",
          "title": "Building a Tumblr theme with GruntJS",
        },
        Object {
          "link": "https://rolflekang.com/backup-routines-for-redisdb",
          "pubDate": "2013-10-24T00:00:00.000Z",
          "title": "Backup routine for Redis",
        },
        Object {
          "link": "https://rolflekang.com/django-app-custom-user",
          "pubDate": "2013-05-22T00:00:00.000Z",
          "title": "Making your Django app ready for custom users",
        },
        Object {
          "link": "https://rolflekang.com/stats-are-fun",
          "pubDate": "2013-03-03T00:00:00.000Z",
          "title": "Stats are fun",
        },
        Object {
          "link": "https://rolflekang.com/alfred-scripts-for-jekyll",
          "pubDate": "2013-02-26T00:00:00.000Z",
          "title": "Alfred scripts for Jekyll",
        },
        Object {
          "link": "https://rolflekang.com/use-array-in-postgresql",
          "pubDate": "2013-01-27T00:00:00.000Z",
          "title": "Use array in postgresql with django",
        },
        Object {
          "link": "https://rolflekang.com/apply-puppet-configuration-automatically",
          "pubDate": "2013-01-03T00:00:00.000Z",
          "title": "Apply puppet automatically",
        },
        Object {
          "link": "https://rolflekang.com/readable-wikipedia",
          "pubDate": "2012-12-02T00:00:00.000Z",
          "title": "Readable Wikipedia",
        },
        Object {
          "link": "https://rolflekang.com/django-nopassword",
          "pubDate": "2012-09-28T00:00:00.000Z",
          "title": "django-nopassword",
        },
        Object {
          "link": "https://rolflekang.com/the-github-lamp",
          "pubDate": "2012-09-16T00:00:00.000Z",
          "title": "The Github lamp",
        },
      ],
      "link": "https://rolflekang.com",
      "title": "Writing by Rolf Erik Lekang",
    },
  },
}
`;

exports[`
  feed(url: "https://rolflekang.com/feed.xml") { title }
 1`] = `
Object {
  "data": Object {
    "feed": Object {
      "title": "Writing by Rolf Erik Lekang",
    },
  },
}
`;

exports[`
  feed(url: "https://rolflekang.com/feed.xml") { title, badField }
 1`] = `
Object {
  "errors": Array [
    Object {
      "extensions": Object {
        "code": "INTERNAL_SERVER_ERROR",
      },
      "message": "Internal server error",
    },
  ],
}
`;

exports[`
  feed(url: "https://rolflekang.com/feed.xml", parser: FEEDME) { title }
 1`] = `
Object {
  "data": Object {
    "feed": Object {
      "title": "Writing by Rolf Erik Lekang",
    },
  },
}
`;

exports[`
  feed(url: "https://rolflekang.com/feed.xml", parser: FEEDPARSER) { title }
 1`] = `
Object {
  "data": Object {
    "feed": Object {
      "title": "Writing by Rolf Erik Lekang",
    },
  },
}
`;

exports[`
  feed(url: "https://rolflekang.com/feed.xml", parser: RSS_PARSER) { title }
 1`] = `
Object {
  "data": Object {
    "feed": Object {
      "title": "Writing by Rolf Erik Lekang",
    },
  },
}
`;

exports[`
  feed(url: "https://rolflekang.com/feed.xml", parser: RSS_TO_JSON) { title }
 1`] = `
Object {
  "data": Object {
    "feed": Object {
      "title": "Writing by Rolf Erik Lekang",
    },
  },
}
`;

exports[`
  feed(url: "https://rolflekang.com/feed.xml", startTime: "2020-01-01", endTime: "2020-10-31") {
    entries { title }
  }
 1`] = `
Object {
  "data": Object {
    "feed": Object {
      "entries": Array [
        Object {
          "title": "Using Ansible handlers in loops",
        },
        Object {
          "title": "Serving text/plain for curl with Next",
        },
        Object {
          "title": "Wireless uplinks with Unifi",
        },
      ],
    },
  },
}
`;

exports[`
  feed(url: "https://rolflekang.com/testing-simple-graphql-services") { title }
 1`] = `
Object {
  "data": Object {
    "feed": null,
  },
  "errors": Array [
    Object {
      "error": Object {
        "code": "not-a-feed",
        "message": "Not a feed",
      },
      "path": Array [
        "feed",
      ],
    },
  ],
}
`;

exports[`
  findFeed(url: "https://rolflekang.com") {
    link
  }
 1`] = `
Object {
  "data": Object {
    "findFeed": Array [
      Object {
        "link": "https://rolflekang.com/feed.xml",
      },
      Object {
        "link": "https://rolflekang.com/feed.json",
      },
    ],
  },
}
`;
